### PR TITLE
Added test cases for Format, ProgressBar, Reader pkgs.

### PR DESCRIPTION
**Pull request by joeyyy09**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

This PR adds test cases for the `formatInfo`, `NewTransport`, and `NewReader` functions in the `pkg/utils/progressbar` package.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: pkg/utils/progressbar/format_test.go [hunk 1]
```diff
@@ -72,3 +74,91 @@ func Test_formatInfo(t *testing.T) {
 		})
 	}
 }
+
+// TestNewTransport verifies the NewTransport function.// TestNewTransport verifies the NewTransport function.
+func TestNewTransport(t *testing.T) {
```
Comment (wzshiming) [inline pkg/utils/progressbar/format_test.go:79]: move to progressbar_test.go

File: pkg/utils/progressbar/format_test.go [hunk 2]
```diff
@@ -72,3 +74,91 @@ func Test_formatInfo(t *testing.T) {
...
+		if resp != nil && resp.Body != nil {
+			if err := resp.Body.Close(); err != nil {
+				t.Errorf("Error closing response body: %v", err)
+			}
+		}
+	}()
+}
+
+// TestNewReader verifies the NewReader function.
+func TestNewReader(t *testing.T) {
```
Comment (wzshiming) [inline pkg/utils/progressbar/format_test.go:103]: move to reader_test.go

File: pkg/utils/progressbar/format_test.go [hunk 3]
```diff
@@ -72,3 +74,91 @@ func Test_formatInfo(t *testing.T) {
...
+	// As reader is already of type io.Reader, no need for type assertion
+	// We can directly use it as an io.Reader
+	_, err := reader.Read(make([]byte, 0)) // Perform a read operation to ensure reader is functional
+	if err != nil {
+		t.Errorf("NewReader did not return a functional reader: %v", err)
+	}
+}
+
+// TestNewReadCloser verifies the NewReadCloser function.
+func TestNewReadCloser(t *testing.T) {
```
Comment (wzshiming) [inline pkg/utils/progressbar/format_test.go:121]: move to reader_test.go

**Review by wzshiming — COMMENTED** [review]
These tests did not actually test any features.

**Comment by joeyyy09** [comment]
Hey @wzshiming I've added the test files for the format, progressbar, reader.go files. Can you please have a look at them?

File: pkg/utils/progressbar/progressbar_test.go [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+// TestTransportRoundTrip verifies the RoundTrip method of the transport.
+func TestTransportRoundTrip(t *testing.T) {
+	// Mock base RoundTripper
+	base := &mockRoundTripper{}
+
+	// Create new transport with progress bar
+	transport := NewTransport(base)
+
+	// Create a sample HTTP request
+	req, err := http.NewRequest(http.MethodGet, "https://example.com", nil)
```
Comment (wzshiming) [inline pkg/utils/progressbar/progressbar_test.go:56]: import "net/http/httptest" and use `httptest.NewServer()` to create a test server

Comment (joeyyy09) [reply pkg/utils/progressbar/progressbar_test.go]: Okay, thanks!

**Comment by joeyyy09** [comment]
@wzshiming I've made the required changes and also squashed all the commits. Can you take a look at them?

**Pull request closed without merging**